File: base-class-module/src/main/java/jingya/com/base_class_module/BaseUtils/TimeTransformUtils.java
package jingya.com.base_class_module.BaseUtils;

import android.text.TextUtils;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

/**
 * @author kuky
 * @description
 */

public class TimeTransformUtils {
    /**
     * 时间戳转换成日期格式字符串
     *
     * @param seconds 精确到秒的字符串
     * @param format
     * @return
     */
    public static String millsTimeStamp2Date(String seconds, String format) {
        if (seconds == null || seconds.isEmpty() || seconds.equals("null")) {
            return "";
        }
        if (TextUtils.isEmpty(format)) format = "yyyy-MM-dd HH:mm:ss";
        SimpleDateFormat sdf = new SimpleDateFormat(format, Locale.getDefault());
        return sdf.format(new Date(Long.valueOf(seconds + "000")));
    }

    /**
     * 时间戳转换成日期格式字符串
     *
     * @param mills  精确到毫秒的字符串
     * @param format
     * @return
     */
    public static String timeStamp2Date(long mills, String format) {
        if (TextUtils.isEmpty(format)) format = "yyyy-MM-dd HH:mm:ss";
        SimpleDateFormat sdf = new SimpleDateFormat(format, Locale.getDefault());
        return sdf.format(new Date(mills));
    }

    /**
     * 日期格式字符串转换成时间戳
     *
     * @param date   字符串日期
     * @param format 如：yyyy-MM-dd HH:mm:ss
     * @return
     */
    public static String date2TimeStamp(String date, String format) {
        try {
            SimpleDateFormat sdf = new SimpleDateFormat(format, Locale.getDefault());
            return String.valueOf(sdf.parse(date).getTime() / 1000);
        } catch (Exception e) {
            e.printStackTrace();
        }
        return "";
    }

    public static String date2String(Date date, String format) {
        if (TextUtils.isEmpty(format)) format = "yyyy-MM-dd HH:mm:ss";
        SimpleDateFormat sdf = new SimpleDateFormat(format, Locale.getDefault());
        return sdf.format(date);
    }
}

File: base-kt-module/src/main/java/com/kuky/base_kt_module/BaseUtils/TimeTransformUtils.kt
package com.kuky.baselib.baseUtils

import android.text.TextUtils
import java.text.SimpleDateFormat
import java.util.*

/**
 * @author kuky
 * @description
 */
object TimeTransformUtils {

    fun timeStamp2Date(seconds: String, format: String = "yyyy-MM-dd HH:mm:ss"): String {
        val sdf = SimpleDateFormat(format, Locale.getDefault())
        return if (TextUtils.isEmpty(seconds)) ""
        else sdf.format(Date(java.lang.Long.valueOf("${seconds}000")))
    }

    fun millsTimeStamp2Date(mills: String, format: String = "yyyy-MM-dd HH:mm:ss"): String {
        val sdf = SimpleDateFormat(format, Locale.getDefault())
        return if (TextUtils.isEmpty(mills)) ""
        else sdf.format(Date(java.lang.Long.valueOf(mills)))
    }

    fun date2TimeStamp(date: String, format: String = "yyyy-MM-dd HH:mm:ss"): String {
        val sdf = SimpleDateFormat(format, Locale.getDefault())
        return (sdf.parse(date).time / 1000).toString()
    }

    fun date2String(date: Date, format: String = "yyyy-MM-dd HH:mm:ss"): String {
        val sdf = SimpleDateFormat(format, Locale.getDefault())
        return sdf.format(date)
    }
}